src/mlpack/tests/one_hot_encoding_test.cpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * @file one_hot_encoding_test.cpp
+ * @author Jeffin Sam
+ *
+ * Tests for onehotencoding().
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <sstream>
+
+#include <mlpack/core.hpp>
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::data;
+using namespace std;
+
+BOOST_AUTO_TEST_SUITE(OneHotEncoding);
+
+/**
+ * Test one hot encoding.
+ */
+BOOST_AUTO_TEST_CASE(OneHotEncodingTest)
+{
+  arma::Mat<size_t> matrix;
+  matrix = "1 0;"
+           "0 1;"
+           "1 0;"
+           "1 0;"
+           "1 0;"
+           "1 0;"
+           "0 1;"
+           "1 0;";
+// Output matrix to save onehotencoding results.
+  arma::Mat<size_t> output;
+  matrix = matrix.t();
+  arma::irowvec labels("-1 1 -1 -1 -1 -1 1 -1");
+  data::OneHotEncoding(labels, output);
+
+  BOOST_REQUIRE_EQUAL(matrix.n_cols, output.n_cols);
+  BOOST_REQUIRE_EQUAL(matrix.n_rows, output.n_rows);
+  CheckMatrices(output, matrix);
+}
+
+BOOST_AUTO_TEST_CASE(OneHotEncodingSparseMatTest)
```

Comment:
Looking at the `OneHotEncodingTest` test case, maybe we should write a template method to avoid some code duplication?

Reply:
There isn't much code duplication, apart from the checks of matrix, which is trivial. 

If you still feel we should go with the template version, just let me know